src/main/java/duke/data/task/Task.java

```diff
@@ -0,0 +1,54 @@
+package duke.data.task;
+
+/**
+ * Class that represents a task.
+ *
+ * @author Won Ye Ji
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructor for Task class.
+     *
+     * @param description Description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Retrieves the done status of the task.
+     *
+     * @return string representation of the done status of the task.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```

Comment:
I think the parentheses are redundant?
```suggestion
        return isDone ? "X" : " "; // mark done task with X
```